modin/numpy/arr.py

```diff
@@ -1545,8 +1549,68 @@ def astype(self, dtype, order="K", casting="unsafe", subok=True, copy=True):
             return self
         return array(_query_compiler=result, _ndim=self._ndim)
 
+    def _build_repr_array(self):
+        # We want to rely on NumPy for creating a string representation of this array; however
+        # we also don't want to materialize all of the data to the head node. Instead, we will
+        # materialize enough data that NumPy can build the summarized representation of the array
+        # (while changing with the NumPy print options so it will format this smaller array as
+        # abridged) and return this smaller array. In the worst case, this array will have
+        # (2*numpy.get_printoptions()["edgeitems"] + 1)^2 items, so 49 items max for the default
+        # value of 3.
+        num_elements = numpy.get_printoptions()["edgeitems"]
```

Comment:
I wonder if we should do `+1` here to account for middle element being removed instead of reducing the threshold down below, so that _shown_ elements' count is correct.
As right now, if I understand correctly, we'd gather N*2 elements (where N is the threshold), and then show N*2-1 elements, while numpy would show N*2?..

Reply:
So we do add +1. The reason we have to modify threshold is because if the size (so total number of elements in the array) of the array is less than the threshold, it will just print the entire array. We need to reduce the threshold to get the `...` numbers if the array is bigger than the threshold.

E.g. the default threshold is 1000. If we have a 2000 long array, and want 3 edge items, we'd get 49 items, and try to repr that. But when we try and get the repr of the 49 items, NumPy thinks that we're trying to repr an array of size 49, so it prints the whole thing out. We need to modify the repr so NumPy still prints the `...` etc. even though it thinks its actually dealing with a relatively smaller array.

To get the `...`, we do the logic above, where if the axis is long enough, we get 0...num_elements and len(self)-num_elements...num_elements elements.